integration/test/Test/LegalHold.hs

```diff
@@ -274,3 +280,522 @@ testLHDeleteClientManually = do
+
+            let sendMessageFromBobToAlice :: HasCallStack => (String -> [String]) -> (Response -> App ()) -> App ()
+                sendMessageFromBobToAlice recipients assertion =
+                  for_ ((,) <$> mbConvId <*> mbLHDevice) \(convId, device) -> do
+                    successfulMsgForOtherUsers <-
+                      mkProteusRecipients
+                        bob -- bob is the sender
+                        [(alice, recipients device), (bob, [bobc])]
+                        -- we send to clients of alice, maybe the legalhold device
+                        -- we need to send to our other clients (bobc)
+                        "hey alice (and eve)" -- the message
+                    let bobaliceMessage =
+                          Proto.defMessage @Proto.QualifiedNewOtrMessage
+                            & #sender . Proto.client .~ (bobc2 ^?! hex)
+                            & #recipients .~ [successfulMsgForOtherUsers]
+                            & #reportAll .~ Proto.defMessage
+                    -- make sure that `convId` is not just the `convId` but also
+                    -- contains the domain because `postProteusMessage` will take the
+                    -- comain from the `convId` json object
+                    postProteusMessage bob convId bobaliceMessage
+                      `bindResponse` assertion
+
+            sendMessageFromBobToAlice (\device -> [alicec, device]) \resp -> do
+              resp.status `shouldMatchInt` 404
+
+            -- now we disable legalhold
+            doDisableLH
+
+            for_ mbLHDevice \lhd ->
+              local (setTimeoutTo 90) $
+                awaitNotification alice alicec noValue isUserClientRemoveNotif >>= \notif ->
+                  notif %. "payload.0.client.id" `shouldMatch` lhd
+
+            let assertStatusFor user status =
+                  getConnections user `bindResponse` \resp -> do
+                    resp.status `shouldMatchInt` 200
+                    conn <- assertOne =<< do resp.json %. "connections" & asList
+                    conn %. "status" `shouldMatch` status
+
+            if testPendingConnection
+              then do
+                assertStatusFor alice "sent"
+                assertStatusFor bob "pending"
+              else do
+                assertStatusFor alice "accepted"
+                assertStatusFor bob "accepted"
+
+            for_ [(alice, alicec, lastNotifAlice'), (bob, bobc, lastNotifBob')] \(user, client, lastNotif) -> do
+              awaitNotification user client (Just lastNotif) isUserConnectionNotif >>= \notif ->
+                notif %. "payload.0.connection.status" `shouldMatchOneOf` ["sent", "pending", "accepted"]
+
+            sendMessageFromBobToAlice (const [alicec]) \resp -> do
+              resp.status `shouldMatchInt` 201
+
+            sendMessageFromBobToAlice (\device -> [device]) \resp -> do
+              resp.status `shouldMatchInt` 412
```

Comment:
```suggestion
            sendMessageFromBobToAlice [alicec] \resp -> do
              resp.status `shouldMatchInt` 201

            sendMessageFromBobToAlice (maybeToList mbLHDevice) \resp -> do
              resp.status `shouldMatchInt` 412
```

...  and further up:

```
            sendMessageFromBobToAlice (alicec : (maybeToList mbLHDevice)) \resp -> do
              resp.status `shouldMatchInt` 404

```

if that can be implemented, i would find it much more readable.